### ISSUE TITLE
[3.3.5] Spell/Shaman: Grounding Totem not work properly

### ISSUE THREAD

**Issue by Noryad** [TrinityCore/TrinityCore#19208]
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

**Description:**
http://wotlk.openwow.com/spell=8177
>Grounding Totem: Summons a Grounding Totem with 5 health at the feet of the caster **that will redirect one harmful spell cast on a nearby party member to itself, destroying the totem**. Will not redirect area of effect spells. Lasts 45 sec.

Grounding Totem works ok if it is summoned before any negative spell. Spells with a travel time (such as Shadowbolt, Firebolt, ...) do not damage the target until the projectile hits. The totem must redirect the spell until it takes damage and summarily gets destroyed. In case there is no totem, if the spell is casted and it is in the air and quickly the shaman uses the Grounding Totem, two things happen:

- First: The spell is never redirected and hits the shaman, totem persist. If the correct thing is that when the spell is casted and there is no totem the spell hits the shaman then this is not a bug, else after the totem is used a check is required.
![case 1](https://cloud.githubusercontent.com/assets/30839/23340257/e456aeae-fbe7-11e6-8404-5deee8c349b2.jpg)
- Second: (After first case) The shaman's opponent casts spells for a few seconds and all spells hits the shaman. Grounding totem ignores spells for a while.
![case 2](https://cloud.githubusercontent.com/assets/30839/23340228/95022d10-fbe7-11e6-9447-d99d59e0a342.jpg)

**Current behaviour:** 
Explained above.

**Expected behaviour:** 
Grounding Totem should redirect spells in all cases.

**Steps to reproduce the problem:**
1. There is no totem
2. Cast shadowbolt to shaman (Better at greater distance).
3. Quickly the shaman uses the Grounding Totem.
4. Shadowbolt is not redirected (first case) and Grounding totem persist.
5. Spam Shadowbolt.
6. Grounding totem ignores spells for a while (second case).

**Branch(es):** 3.3.5

**TC rev. hash/commit:**  4224259

**TDB version:**  TDB_full_world_335.62_2016_10_17 + updates

**Operating system:**  Windows

[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)
[//]: # (For fixes containing C++, create a Pull Request.)

**Comment by robinsch** [comment]
Redirect was always checked on cast.

**Comment by Moorgoth** [comment]
hmmm, then something is wrong, confirmed for me

**Comment by Noryad** [comment]
@ariel- can you take a look here?

**Comment by Noryad** [comment]
7072bf8 solves the second case what about not the first?

**Comment by ariel-** [comment]
Blizzlike

**Comment by Killyana** [comment]
I just updated the core+DB and I got this DB errors:
```
The `spell_proc` table entry for spellId 8178 doesn't have any `ProcFlags` value defined, proc will not be triggered.
The `spell_proc` table entry for spellId 8178 has `SpellTypeMask` value defined, but it will not be used for the defined `ProcFlags` value.
The `spell_proc` table entry for spellId 38857 doesn't have any `ProcFlags` value defined, proc will not be triggered.
The `spell_proc` table entry for spellId 38857 has `SpellTypeMask` value defined, but it will not be used for the defined `ProcFlags` value.

**Comment by ariel-** [comment]
Fixed in ebba8dcba87190cfdef55bd896a11d2d234cd754, thanks
